### PR TITLE
Allow several alerts in single replica topology

### PR DESCRIPTION
**Pull request by omertuc**
EDIT:
This PR used to be just for [MON-1522](https://issues.redhat.com/browse/MON-1522), but now it's also for  https://bugzilla.redhat.com/show_bug.cgi?id=1945017 and  [API-1136](https://issues.redhat.com/browse/API-1136)

### CODE REVIEW

**Comment by romfreiman** [comment]
/lgtm
Can u please connect it to the jira ticket as a workaround?

**Comment by romfreiman** [comment]
@smarterclayton wdyt? Temporary till it will be fixed on the product level. Otherwise we cannot enable sno ci

**Comment by omertuc** [comment]
> Can u please connect it to the jira ticket as a workaround?

That's done automatically when the issue is referenced, but commented anyway

File: test/extended/prometheus/prometheus.go [hunk 1]
```diff
@@ -396,11 +406,20 @@ var _ = g.Describe("[sig-instrumentation] Prometheus", func() {
 				oc.AdminKubeClient().CoreV1().Pods(ns).Delete(context.Background(), execPod.Name, *metav1.NewDeleteOptions(1))
 			}()
 
+			// Checking Watchdog alert state is done in "should have a Watchdog alert in firing state".
+			allowedAlerts := []string{"Watchdog", "AlertmanagerReceiversNotConfigured", "PrometheusRemoteWriteDesiredShards"}
+
+			infra, err := oc.AdminConfigClient().ConfigV1().Infrastructures().Get(context.Background(), "cluster", metav1.GetOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+			if infra.Status.ControlPlaneTopology == configv1.SingleReplicaTopologyMode {
```
Comment (smarterclayton) [inline test/extended/prometheus/prometheus.go:414]: move these up to the top and give them unique names “allowedAlertsDuringX”, don’t reuse the same variable

Comment (omertuc) [reply test/extended/prometheus/prometheus.go:414]: By top do you mean the top of the `It` block?

Also not sure how I can avoid reusing the same variable without duplicating the Prometheus query's `fmt.Sprintf` - it has to use some variable for `strings.Join(allowedAlerts, "|")`

Comment (smarterclayton) [reply test/extended/prometheus/prometheus.go:414]: Nm, I thought you were fixing the upgrade job which has both of these inline.

Comment (smarterclayton) [reply test/extended/prometheus/prometheus.go:414]: Move the infra config lookup up above allowedAlerts, then init allowedAlerts, then do the if check.  Roughly keep infra lookup high and the init of allowedAlerts and the conditional close to each other.

Comment (omertuc) [reply test/extended/prometheus/prometheus.go:415]: Got it, done

/hold until I verify this works (takes a while)

Comment (omertuc) [reply test/extended/prometheus/prometheus.go:415]: /unhold

**Comment by omertuc** [comment]
/test e2e-aws-single-node

this time with m5d.4xlarge

**Comment by omertuc** [comment]
These alerts have been fixed by now, PR no longer relevant

**Pull request closed without merging**